crates/goose/src/session/legacy.rs

```diff
@@ -0,0 +1,113 @@
+use crate::conversation::Conversation;
+use crate::session::Session;
+use anyhow::Result;
+use chrono::NaiveDateTime;
+use std::fs;
+use std::io::{self, BufRead};
+use std::path::{Path, PathBuf};
+use std::time::SystemTime;
+
+const MAX_FILE_SIZE: u64 = 50 * 1024 * 1024;
+
+pub fn list_sessions(session_dir: &PathBuf) -> Result<Vec<(String, PathBuf)>> {
+    let entries = fs::read_dir(session_dir)?
+        .filter_map(|entry| {
+            let entry = entry.ok()?;
+            let path = entry.path();
+
+            if path.extension().is_some_and(|ext| ext == "jsonl") {
+                let name = path.file_stem()?.to_string_lossy().to_string();
+                Some((name, path))
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+
+    Ok(entries)
+}
+
+pub fn load_session(session_name: &str, session_path: &Path) -> Result<Session> {
+    let file = fs::File::open(session_path).map_err(|e| {
+        anyhow::anyhow!(
+            "Failed to open session file {}: {}",
+            session_path.display(),
+            e
+        )
+    })?;
+
+    let file_metadata = file.metadata()?;
+
+    if file_metadata.len() > MAX_FILE_SIZE {
+        return Err(anyhow::anyhow!("Session file too large"));
+    }
+    if file_metadata.len() == 0 {
+        return Err(anyhow::anyhow!("Empty session file"));
+    }
+
+    let modified_time = file_metadata.modified().unwrap_or(SystemTime::now());
+    let created_time = file_metadata
+        .created()
+        .unwrap_or_else(|_| parse_session_timestamp(session_name).unwrap_or(modified_time));
```

Comment:
were the old session names human readable title strings? or is this a different field from the metadata and called `session_name` in this code